frontend/src/employee-frontend/components/vasu/components/FollowupQuestion.tsx

```diff
@@ -41,7 +41,17 @@ const FollowupEntryElement = React.memo(function FollowupEntryElement({
     [onEdited]
   )
 
-  const editAllowed = user?.id === entry.authorId
+  const getEntry = useCallback(
+    () => (editedEntry ? editedEntry : entry),
+    [entry, editedEntry]
+  )
+
+  const editAllowed =
+    user?.id === entry.authorId ||
+    roles.includes('ADMIN') ||
+    roles.includes('UNIT_SUPERVISOR') ||
+    roles.includes('STAFF') ||
+    roles.includes('SPECIAL_EDUCATION_TEACHER')
```

Comment:
New role-based checks in the frontend 😬 This should be changed to use `getPermittedActions` in the service and send the results to frontend somehow.